src/open_company_web/components/topics_columns.cljs

```diff
@@ -203,38 +229,53 @@
           ;; render 2 or 3 column layout
           (> columns-num 1)
           (dom/div {:class "topics-column-container group"
-                    :style #js {:width total-width}}
+                    :style #js {:width total-width}
+                    :key columns-container-key}
             (for [kw (if (= columns-num 3) [:1 :2 :3] [:1 :2])]
-              (let [column (get best-layout kw)]
-                (dom/div {:class "topics-column"
-                          :style #js {:width (str card-width "px")}}
-                  (when (pos? (count column))
-                    (for [idx (range (count column))
-                      :let [section-kw (get column idx)
-                            section-name (name section-kw)]]
-                      (partial-render-topic section-name
-                                            (when (= section-name "add-topic") (int (name kw))))))
-                  (when (and show-add-topic
-                             (= kw :1)
-                             (= (count topics) 0))
-                    (partial-render-topic "add-topic" 1))
-                  (when (and show-add-topic
-                             (= kw :2)
-                             (or (and (= (count topics) 1)
-                                      (= columns-num 3))
-                                 (and (>= (count topics) 1)
-                                      (= columns-num 2))))
-                    (partial-render-topic "add-topic" 2))
-                  (when (and show-add-topic
-                             (= kw :3)
-                             (>= (count topics) 2))
-                    (partial-render-topic "add-topic" 3))))))
+              (let [column (get best-layout kw)
+                    {:keys [pinned other]} (utils/get-pinned-other-keys column (dis/company-data))]
+                (dom/div {:class (str "topics-column col-" (name kw))
+                          :style #js {:width (str (+ card-width 20 -8) "px")}}
+                  (dom/div #js {:className "topics-column-pinned"}
```

Comment:
Some comments in here explaining the significance of these when conditionals would be good.